src/cudamatrix/cu-device.cc

```diff
@@ -109,9 +109,10 @@ void CuDevice::Initialize() {
     cudaSetDevice(device_id_);
     // Initialize CUBLAS.
     CUBLAS_SAFE_CALL(cublasCreate(&cublas_handle_));
+    CUBLAS_SAFE_CALL(cublasSetStream(cublas_handle_, cudaStreamPerThread));
```

Comment:
@galv, thanks.  I believe justin mentioned that we also need to use this cudaStreamPerThread for other API calls, like memcpy.